enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+
+#### Story 1
+
+At the end of phase one, the SNC image build process will be slightly
+simpler, and should run a bit faster. The resulting CRC cluster that
+end users deploy should be unchanged.
+
+#### Story 2
+
+As phase two progresses with merges to various components, the SNC
+image build process will become even simpler and quicker, and the
+resulting CRC cluster should require less system resources.
+
+### Implementation Details/Notes/Constraints
+
+Phase 1 is pretty straightforward - simply adding an annotation to the
+existing manifest files in the various repositories for components to
+be included in single node developer clusters.
+
+Phase 2 is more of an open ended process leveraging the new profile to
+optimize the configuration of the individual components for single
+node developer clusters. In general, each component will be addressed
+independently and possibly multiple times, first to eliminate the
+post-install patching by SNC, and then to enable additional
+optimizations. During implementation of this phase, it is expected
+that patterns will emerge when common approaches apply across
+different components.
+
+### Risks and Mitigations
+
+One risk is that all parts of the implementation of the cluster
+profiles
+[enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md)
+do not get merged in a timely manner. Implementing the
+single-node-developer cluster profile requires the ability to specify
+a non-default cluster profile to the OpenShift installer, and for this
+to result in only the manifests annotated for the specified cluster
+profile to be processed by the resulting cluster. Developer tools team
+members have proposed CVO patches as part of that enhancement, and
+will continue to participate in the review process, but progress
+getting the implementation merged has been slow so far. Having this
+enhancement as an additional use case for the cluster profile
+mechanism hopefully will help clarify how that feature is being used
+and expedite its implementation.
+
+Another risk is that it will take too long to get the initial
+annotations merged to all the needed OpenShift component repositories
+during the first phase of development. Until all needed annotations
+are merged, the SNC tooling cannot be updated to make use of the
+single-node-developer cluster profile, as needed components would be
+excluded from the cluster. The review process for this enhancement
+should help raise awareness across the various teams that will need to
+review and merge these patches.
+
+A final risk is that new components will be added to OpenShift after
+the completion of phase 1 that should be included in single node
+developer clusters, but will not have the necessary annotation
+included in their manifests due to lack of awareness or concern for
+the CRC product. Given that single node developer clusters are
+expected to pass the majority of the OpenShift conformance tests,
```

Comment:
I ran `openshift-tests run openshift/conformance` against the 1.17 CodeReady Containers release, and the results were `error: 64 fail, 913 pass, 1660 skip (1h44m11s)`. Half of these are Prometheus/OauthServer/TopologyManager, so probably because of disabled components. Haven't looked closely at the others yet.

Reply:
I did not mean to imply via "majority" that >= 50% of the tests were required to pass. The expectation is that all tests that do not depend on disabled functionality will pass. We will need to configure the CI job to skip those tests that are not expected to pass.

If a patch introducing a new component does not include the single-node-developer profile's annotation on a manifest, but does add new conformance tests for that new component, the CI job should begin to fail, unless the CI job is also configured to skip those new tests. This will help mitigate the risk of the annotation being accidentally omitted from the new component's manifest.

Reply:
@cfergeau thanks for the data.
@soltysh the important requirement from our present state is we have an explicit list of things we _know_ pass so we know what level of functionality is expected to work (versus assume).